src/message.rs

```diff
@@ -21,6 +23,21 @@ fn from_cbor_blake2b256<S: serde::ser::Serialize>(obj: &S) -> Result<Cid, Error>
     ))
 }
 
+pub fn parse_address(s: &str) -> anyhow::Result<Address> {
+    if let Ok(addr) = Network::Testnet
+        .parse_address(s)
+        .or_else(|e| {
+            Network::Mainnet.parse_address(s)
+        })
+    {
+        return Ok(addr);
+    }
+
+    // Try parsing as 0x ethereum address
+    let addr = hex::decode(&s[2..])?;
+    Ok(Address::new_delegated(10, &addr)?)
```

Comment:
This approach is overly simplistic and doesn't handle correct errors and invalid input.

1. If the input length is smaller than 3, it will panic (index out of bounds). We don't want to crash the app, even a serverless one; we want to handle errors gently and display a helpful message.
2. `new_delegated` doesn't perform meaningful checks, except for the max length check. Sending to `0x12` (or any invalid address) will not trigger errors on the faucet side, leaving the user puzzled. I verified this with:
![image](https://github.com/user-attachments/assets/fa28e413-c401-483b-8820-1d96afeedb6d)

as opposed to the current behaviour

![image](https://github.com/user-attachments/assets/006826f5-6fe3-493a-82e3-ec82604575c9)

Please consider adding "unhappy" path test cases and validating it on a local deployment.

Reply:
Fixed some issues and added more tests 👍